src/main/java/org/wso2/carbon/module/swiftiso20022/utils/ValidatorUtils.java

```diff
@@ -18,21 +18,64 @@
 
 package org.wso2.carbon.module.swiftiso20022.utils;
 
+import org.apache.commons.lang3.StringUtils;
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
 import org.wso2.carbon.module.swiftiso20022.model.ErrorModel;
 
+import java.util.Currency;
+import java.util.Set;
+import java.util.regex.Pattern;
+
 /**
  * Class to validate the request payload.
  */
 public class ValidatorUtils {
 
-    /**
-     * Method to construct the error model.
+    /** Method to validate whether currency is in ISO 4217 format
+     *
+     * @param currency  Currency to be validated
+     * @return     Whether currency is valid
+     */
+    public static boolean isValidCurrency(String currency) {
+        if (StringUtils.isBlank(currency)) {
+            return false;
+        }
+        Set<Currency> availableCurrencies = Currency.getAvailableCurrencies();
+        return availableCurrencies.stream().anyMatch(c -> c.getCurrencyCode().equals(currency));
+    }
+
+    /** Method to validate whether amount is valid
+     *
+     * @param amount  Amount to be validated
+     * @return     Whether amount is valid
+     */
+    public static ErrorModel validateAmountLength(String amount, String fieldName) {
+        ErrorModel errorModel = new ErrorModel();
+
+        if (amount == null) {
+            return new ErrorModel(ConnectorConstants.ERROR_T13,
+                    ConnectorConstants.ERROR_AMOUNT_NULL);
+        }
+        if (amount.isBlank()) {
+            return new ErrorModel(ConnectorConstants.ERROR_C03,
+                    ConnectorConstants.ERROR_AMOUNT_SIZE_INVALID);
+        }
+
+        if (amount.length() > 16) {
+            return new ErrorModel(ConnectorConstants.ERROR_M50,
+                    String.format(ConnectorConstants.ERROR_PARAMETER_LENGTH,
+                            fieldName + ConnectorConstants.AMOUNT, 15));
+        }
+        errorModel.setIsError(false);
+        return errorModel;
```

Comment:
```suggestion
        return new ErrorModel();
```

Reply:
FIxed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/befc8f745ba4f7a2cb0b949ac349d98c62e8ec5b